File: Sub_array.py
class Solution:
    def subarraySum(self, nums: list[int], K: int) -> int:
        res = 0
        curSum = 0
        prefixSums = {0: 1}

        for n in nums:
            curSum += n
            diff = curSum - K

            res += prefixSums.get(diff, 0)
            prefixSums[curSum] = 1 + prefixSums.get(curSum, 0)

        return res

File: Subarray.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        
        map<int,int> m;

        m[0] = 1;

        int rsum = 0;
        int count = 0;

        for(int i=0;i<nums.size();i++){

            rsum = rsum + nums[i];

            int diff = rsum - k;

            if(m.find(diff) != m.end()){
                      count = count + m[diff];
              
            }
           
           m[rsum]++;
        }

            return count;

    }
};